src/main/java/org/testng/internal/reflect/ArrayEndingMethodMatcher.java

```diff
@@ -0,0 +1,86 @@
+package org.testng.internal.reflect;
+
+import org.testng.ITestContext;
+import org.testng.ITestResult;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Method;
+
+import static org.testng.internal.reflect.InjectableParameter.Assistant.ALL_INJECTS;
+
+/**
+ * Checks for array ending method argument match with or without filtering injectables.
+ *
+ * @author <a href="mailto:nitin.matrix@gmail.com">Nitin Verma</a>
+ * @see ReflectionRecipes#matchArrayEnding(Class[], Object[])
+ */
+public class ArrayEndingMethodMatcher implements MethodMatcher {
+
+  private final Method method;
+  private final Parameter[] methodParameter;
+  private final ITestContext context;
+  private final ITestResult testResult;
+  private final Object[] userArguments;
+  private Boolean conforms = null;
```

Comment:
point taken and used an enum.

https://github.com/nitinverma/testng/blob/feat-data-providers/src/main/java/org/testng/internal/reflect/ArrayEndingMethodMatcher.java#L27-L27